### PR TITLE
Fixes #17139: add jq to rudder agent

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/17139

### CODE REVIEW

File: rudder-agent/SOURCES/Makefile [hunk 1]
```diff
@@ -570,6 +588,12 @@ build-curl: curl-source
 	cd curl-source && $(MAKE) install DESTDIR=$(CURDIR)/dependencies
 	touch $@
 
+build-jq: jq-source
+	cd jq-source && autoreconf -fi && ./configure --prefix=$(RUDDER_DIR) --enable-gettext=no $(BUILD_FLAGS) --disable-maintainer-mode --disable-docs --with-oniguruma=builtin
```
Comment (peckpeck) [inline rudder-agent/SOURCES/Makefile:592]: what does --enable-gettext=no do ?

File: rudder-agent/SPECS/dependencies [hunk 2]
```diff
@@ -1,9 +1,9 @@
 sles-10sp3:gcc bison flex autoconf automake libtool pcre-devel zlib-devel pam-devel libxml2-devel curl-devel libacl-devel
 sles-11:gcc openssl-devel bison flex autoconf automake libtool pcre-devel zlib-devel libbz2-devel pam-devel libxml2-devel libcurl-devel libacl-devel
 sles-12:gcc openssl-devel bison flex autoconf automake libtool pcre-devel zlib-devel libbz2-devel pam-devel libyaml-devel libxml2-devel curl-devel libacl-devel
-sles-15:gcc openssl-devel bison flex autoconf automake libtool pcre-devel zlib-devel libbz2-devel pam-devel libyaml-devel libxml2-devel curl-devel libacl-devel
+sles-15:gcc openssl-devel bison flex autoconf automake libtool jq pcre-devel zlib-devel libbz2-devel pam-devel libyaml-devel libxml2-devel curl-devel libacl-devel
```
Comment (peckpeck) [inline rudder-agent/SPECS/dependencies:4]: File is used to automatically install packager on the builder, so it is a build dependency.
So you need the -devel version of the package

File: rudder-agent/SPECS/dependencies [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 rhel-3:gcc bison flex autoconf automake libtool pcre-devel make byacc zlib-devel bzip2-devel kernel-utils pam-devel libxml2-devel curl-devel libacl-devel libattr-devel
 rhel-5:gcc openssl-devel bison flex autoconf automake libtool pcre-devel make byacc zlib-devel bzip2-devel pam-devel libxml2-devel curl-devel libacl-devel
 rhel-6:gcc openssl-devel bison flex autoconf automake libtool pcre-devel make byacc pam-devel perl-ExtUtils-MakeMaker perl-IPC-Cmd zlib-devel bzip2-devel libyaml-devel libxml2-devel curl-devel libacl-devel
 rhel-7:gcc openssl-devel bison flex autoconf automake libtool pcre-devel make byacc pam-devel perl-ExtUtils-MakeMaker perl-Digest perl-IPC-Cmd zlib-devel bzip2-devel libyaml-devel libxml2-devel curl-devel libacl-devel
-rhel-8:gcc openssl-devel bison flex autoconf automake libtool pcre-devel make byacc pam-devel perl-ExtUtils-MakeMaker perl-Digest perl-IPC-Cmd zlib-devel bzip2-devel libyaml-devel libxml2-devel curl-devel libacl-devel
+rhel-8:gcc openssl-devel bison flex autoconf automake libtool pcre-devel make byacc pam-devel perl-ExtUtils-MakeMaker perl-Digest perl-IPC-Cmd jq zlib-devel bzip2-devel libyaml-devel libxml2-devel curl-devel libacl-devel
```
Comment (peckpeck) [inline rudder-agent/SPECS/dependencies:9]: Idem

File: rudder-agent/SPECS/rudder-agent.spec [hunk 4]
```diff
@@ -44,6 +44,9 @@
 # We need to build curl since we embed openssl
 %define use_system_curl false
 
+# Same goes for the use of the local jq install vs. a bundled one
+%define use_system_jq true
```
Comment (peckpeck) [inline rudder-agent/SPECS/rudder-agent.spec:48]: You must also declare jq as a dependency here when system_jq is true
And declafe jq-devel as a build-dependency too

File: rudder-agent/debian/control [hunk 5]
```diff
@@ -2,7 +2,7 @@ Source: rudder-agent
 Section: admin
 Priority: extra
 Maintainer: Rudder Team <dev@rudder.io>
-Build-Depends: debhelper (>= 7), bison, gcc, flex, autoconf, automake, libtool, libpcre3-dev, libpam0g-dev, ca-certificates, perl, lsb-release, libyaml-dev, libxml2-dev, libacl1-dev, python, rsync
+Build-Depends: debhelper (>= 7), bison, gcc, flex, autoconf, automake, libtool, libpcre3-dev, libpam0g-dev, ca-certificates, perl, lsb-release, jq (>=9), libyaml-dev, libxml2-dev, libacl1-dev, python, rsync
```
Comment (peckpeck) [inline rudder-agent/debian/control:5]: you need the -dev here since it is a build dependency

Comment (peckpeck) [reply rudder-agent/debian/control:5]: anf why the >=9 ?

Comment (gpoblon) [reply rudder-agent/debian/control:5]: Thought it was debian version.
Fix coming

File: rudder-agent/SPECS/rudder-agent.spec (hunk 4 above)
Comment (gpoblon) [reply rudder-agent/SPECS/rudder-agent.spec:48]: Not sure I fixed it properly,  but "fix coming"

File: rudder-agent/SOURCES/Makefile (hunk 1 above)
Comment (gpoblon) [reply rudder-agent/SOURCES/Makefile:592]: Oh I missed it, should not be there, bad copy/pasting...

File: rudder-agent/SPECS/rudder-agent.spec [hunk 6]
```diff
@@ -281,6 +294,12 @@ BuildRequires: curl-devel
 Requires: curl
 %endif
 
+## JQ dependencies
+%if "%{use_system_jq}" == "true"
+BuildRequires: jq-devel
```
Comment (amousset) [inline rudder-agent/SPECS/rudder-agent.spec:299]: What is it used for?

Comment (gpoblon) [reply rudder-agent/SPECS/rudder-agent.spec:299]: It is what I thought was the fix for @peckpeck comment: 
> You must also declare jq as a dependency here when system_jq is true
> And declafe jq-devel as a build-dependency too

Comment (amousset) [reply rudder-agent/SPECS/rudder-agent.spec:299]: do we build something relying on jq? CFEngine only uses the binary.

Comment (peckpeck) [reply rudder-agent/SPECS/rudder-agent.spec:299]: i thought it was using the library

File: rudder-agent/SPECS/rudder-agent.spec [hunk 7]
```diff
@@ -281,6 +294,12 @@ BuildRequires: curl-devel
...
+Requires: jq
+%endif
+
```
Comment (peckpeck) [inline rudder-agent/SPECS/rudder-agent.spec:302]: I think you forgot to use the use_system_jq variable in the makes commands of this file

File: rudder-agent/SPECS/rudder-agent.spec (hunk 6 above)
Comment (peckpeck) [reply rudder-agent/SPECS/rudder-agent.spec:299]: so : 
- no JQ_ARGS
- no change to cfengine configure
- no change to build-depends

Comment (gpoblon) [reply rudder-agent/SPECS/rudder-agent.spec:299]: removed some things, hopefully the one I had to

File: rudder-agent/SPECS/rudder-agent.spec [hunk 8]
```diff
@@ -129,6 +138,10 @@
 # PIE and PIC incompatible on old gcc
 %define use_pie false
 %endif
+%if 0%{?suse_version} && 0%{?suse_version} < 1500
+# no jq on sles 10 11 and 12
+%define use_system_jq false
+%endif
```
Comment (peckpeck) [inline rudder-agent/SPECS/rudder-agent.spec:144]: you need to use the use_system_jq variable womewhere

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**